packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -42,9 +45,19 @@ abstract class VariantPlatformPlugin {
 }
 
 abstract class NativeOrDartPlugin {
-  /// Determines whether the plugin has a native implementation or if it's a
-  /// Dart-only plugin.
-  bool isNative();
+  /// Determines whether the plugin has a FFI implementation or if it's a
+  /// Dart-only or method channel plugin.
+  ///
+  /// If plugins use both method channels and FFI they should be classified
+  /// as method channel plugins.
+  bool isFfi();
+
+  /// Determines whether the plugin has a method channel implementation or if
+  /// it's a Dart-only or FFI plugin.
+  ///
+  /// If plugins use both method channels and FFI they should be classified
+  /// as method channel plugins.
+  bool isMethodChannel();
```

Comment:
(this applies more broadly than just here)

I am uncomfortable with the way that we're using "ffi plugin" to mean "not a method channel plugin". It seems like an "ffi plugin" should refer to one that uses FFI, regardless of whether it's using method channels or not as well.

In particular, I don't think this scales as we add more types. If the types are not mutually exclusive then IMHO we should treat them as complementary flags. Here we're more or less using two methods returning booleans as an enum, which seems like a model that doesn't actually match the underlying reality.

Reply:
I've updated the docs and implementation.